tools/explorer/lib/model.py

```diff
@@ -47,25 +79,99 @@ class EntityTypeField(StandardField):
   def __init__(self,
                namespace_name: str,
                standard_field_name: str,
-               is_required: bool,
+               is_optional: bool,
                increment: str = ''):
-    """Initializs the EntityTypeField.
-
-    Args:
-       namespace_name: the name of the field's namespace as a string.
-       standard_field_name: the name of the field as a string.
-       is_required: optionality of the field as a boolean.
-       increment: the increment of the field as a string.
-    """
     super().__init__(namespace_name, standard_field_name)
+    if not FIELD_INCREMENT_REGEX.match(increment):
+      raise ValueError(
+          f'Incremement of {namespace_name}/{standard_field_name}{increment} '
+          + 'is unproperly formatted'
+      )
     self._increment = increment
-    self._is_required = is_required
+    self._is_optional = is_optional
+
+  def __hash__(self):
+    return hash((
+        self._namespace,
+        self._name,
+        self._increment,
+        self._is_optional
+    ))
+
+  def __eq__(self, other):
+    if not isinstance(other, self.__class__):
+      raise TypeError(
+          '{str(other)} and {str(self)} must be EntityTypeField objects'
```

Comment:
`EntityTypeField`is a concept that does not exist in our documentation. What does the user need to do when they have this error?

Reply:
@Charbull I'm thinking that a human user won't have to deal with these errors. This is a class created purely for data storage, and if these errors are thrown, then it would be through a mistake made in another module.